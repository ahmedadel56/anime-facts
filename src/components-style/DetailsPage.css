.fa-arrow-circle-left {
  font-size: 5rem;
  position: absolute;
  top: 4rem;
  left: 2rem;
}
