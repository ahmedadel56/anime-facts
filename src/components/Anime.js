import React from 'react';
import PropTypes from 'prop-types';
import { Link, Outlet } from 'react-router-dom';
import { Image, Card } from 'react-bootstrap';

const Anime = (props) => {
  const { id, name, image } = props;
  return (
    <>
      <Link to={`/details/${id}`}>
        <Card style={{ background: 'inherit', marginBottom: '2rem' }}>
          <Image src={image} alt="anime" />
          <Card body>
            <Card text style={{ border: 'none', textAlign: 'center' }}>
              {name}
            </Card>

          </Card>
        </Card>
      </Link>
      <Outlet />
    </>
  );
};

Anime.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};

export default Anime;
