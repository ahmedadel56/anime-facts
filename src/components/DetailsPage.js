import 'bootstrap/dist/css/bootstrap.min.css';
import { Col, Container, Row } from 'react-bootstrap';
import { BsFillSkipBackwardCircleFill } from 'react-icons/bs';
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router';
import { Link } from 'react-router-dom';
import { fetchAnimesDetails } from '../redux/animesDetails';

/* eslint-disable camelcase */

const DetailsPage = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchAnimesDetails());
  }, []);
  const animessState = useSelector((state) => state.anmiesDetailsReducer.animesDetails);
  const { key } = useParams();
  const filteredAnime = animessState.slice(key - 1, key);
  return (
    <Container style={{ minHeight: '100vh' }}>
      {filteredAnime.map((element) => (
        <div key={element.key}>
          <Row>
            <Col md={3}>
              <Link to="/">
                <h1>
                  <BsFillSkipBackwardCircleFill style={{ color: 'white' }} />
                </h1>
              </Link>
            </Col>
            <Col md={6}>
              <img src={element.img} alt="Anime Name" />
            </Col>
          </Row>
          <Row>
            <Col style={{
              display: 'flex', justifyContent: 'center', alignItems: 'center', margin: '2rem 0', color: 'white',
            }}
            >
              <h2>Total Facts : </h2>
              <h3>{element.total_facts}</h3>
            </Col>
          </Row>
          <Container style={{ border: '0.2rem solid white', borderRadius: '30px 30px 0 0' }}>

            {element.data.map((fact) => (
              <Row key={fact.fact_id} style={{ borderBottom: '2px solid white' }}>
                <Col md={4} style={{ textAlign: 'end', color: 'white', fontWeight: '800' }}>
                  <h5>
                    {' '}
                    {'Fact number '}
                    {fact.fact_id}
                    { ': '}
                  </h5>
                </Col>
                <Col>
                  <p style={{ fontSize: '20px', fontWeight: '500' }}>
                    {fact.fact}
                  </p>
                </Col>
              </Row>
            ))}
          </Container>
        </div>
      ))}
    </Container>
  );
};

export default DetailsPage;
