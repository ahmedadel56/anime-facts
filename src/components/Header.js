import React from 'react';
import { Image, Col, Row } from 'react-bootstrap';
import Logo from '../assets/logo.jpeg';

const Header = () => (
  <Row style={{ background: '#588cef', color: 'white', marginBottom: '3rem' }}>
    <Col style={{ paddingLeft: '0' }}><Image src={Logo} alt="Logo Image" /></Col>
    <Col style={{
      textAlign: 'center', display: 'flex', flexDirection: 'column', justifyContent: 'center',
    }}
    >
      <h1>ANIMES FACTS</h1>
      <h2>13 awesome Animes </h2>
    </Col>
  </Row>
);

export default Header;
