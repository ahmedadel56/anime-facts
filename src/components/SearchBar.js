import React from 'react';
import { Form } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { filterAnimes } from '../redux/animes';

const SearchBar = () => {
  const dispatch = useDispatch();
  const filter = (event) => {
    dispatch(filterAnimes(event.target.value));
  };
  return (
    <>

      <Form.Control
        size="lg"
        type="text"
        placeholder="Search for Anime"
        onChange={(event) => {
          filter(event);
        }}
        style={{ marginBottom: '1rem' }}

      />
    </>
  );
};

export default SearchBar;
