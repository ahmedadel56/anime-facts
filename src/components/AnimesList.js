import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchAnimes } from '../redux/animes';
import Anime from './Anime';

const AnimesList = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchAnimes());
  }, []);
  const AnmiesState = useSelector((state) => state.anmiesReducer.animes);

  return (
    <div>
      {AnmiesState.map((anime) => (
        <Anime
          key={anime.anime_id}
          facts={anime.total_facts}
          image={anime.img}
        />
      ))}
    </div>
  );
};

export default AnimesList;
