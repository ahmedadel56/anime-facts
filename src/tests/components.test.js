import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import DetailsPage from '../components/DetailsPage';
import Header from '../components/Header';
import AnimesList from '../components/AnimesList';
import store from '../redux/configureStore';
import Anime from '../components/Anime';
import SearchBar from '../components/SearchBar';

const MockObj = {
  name: 'test',
  image: 'imange',
  id: 123,
};

describe('Detailed Anime component', () => {
  test('matches Header component snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render filterd Anime', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <DetailsPage />
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  test('render filterd Anime', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <AnimesList />
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('render filterd Anime', () => {
    const component = renderer
      .create(
        <BrowserRouter>
          <Anime
            key={MockObj.id}
            id={MockObj.id}
            name={MockObj.name}
            image={MockObj.image}
          />
        </BrowserRouter>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('render Search Bar', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <SearchBar />
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
