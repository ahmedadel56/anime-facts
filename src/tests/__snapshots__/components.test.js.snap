// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detailed Anime component matches Header component snapshot 1`] = `
<div
  className="row"
  style={
    Object {
      "background": "#588cef",
      "color": "white",
      "marginBottom": "3rem",
    }
  }
>
  <div
    className="col"
    style={
      Object {
        "paddingLeft": "0",
      }
    }
  >
    <img
      alt="Logo Image"
      className=""
      src="logo.jpeg"
    />
  </div>
  <div
    className="col"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    <h1>
      ANIMES FACTS
    </h1>
    <h2>
      13 awesome Animes 
    </h2>
  </div>
</div>
`;

exports[`Detailed Anime component render Search Bar 1`] = `
<input
  className="form-control form-control-lg"
  onChange={[Function]}
  placeholder="Search for Anime"
  style={
    Object {
      "marginBottom": "1rem",
    }
  }
  type="text"
/>
`;

exports[`Detailed Anime component render filterd Anime 1`] = `
<div
  className="container"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
/>
`;

exports[`Detailed Anime component render filterd Anime 2`] = `
<div
  className="container"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <div
    className="row"
    style={
      Object {
        "background": "#588cef",
        "color": "white",
        "marginBottom": "3rem",
      }
    }
  >
    <div
      className="col"
      style={
        Object {
          "paddingLeft": "0",
        }
      }
    >
      <img
        alt="Logo Image"
        className=""
        src="logo.jpeg"
      />
    </div>
    <div
      className="col"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      <h1>
        ANIMES FACTS
      </h1>
      <h2>
        13 awesome Animes 
      </h2>
    </div>
  </div>
  <input
    className="form-control form-control-lg"
    onChange={[Function]}
    placeholder="Search for Anime"
    style={
      Object {
        "marginBottom": "1rem",
      }
    }
    type="text"
  />
  <div
    className="row"
  />
</div>
`;

exports[`Detailed Anime component render filterd Anime 3`] = `
<a
  href="/details/123"
  onClick={[Function]}
>
  <div
    className="card"
    style={
      Object {
        "background": "inherit",
        "marginBottom": "2rem",
      }
    }
  >
    <img
      alt="anime"
      className=""
      src="imange"
    />
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="card text-true"
          style={
            Object {
              "border": "none",
              "textAlign": "center",
            }
          }
        >
          test
        </div>
      </div>
    </div>
  </div>
</a>
`;
